Zend/tests/final_properties/access/property_by_ref_access3.phpt

```diff
@@ -0,0 +1,32 @@
+--TEST--
+Test that declared static final properties can't be accessed by reference
+--XFAIL--
+Final static properties can still be accessed
+--FILE--
+<?php
+
+class Foo
+{
+    final public static int $property1;
+    final public static int $property2 = 2;
+
+    public function __construct()
+    {
+        $this->property1 = 1;
+    }
+}
+
+$foo = new Foo();
+
+try {
+    foreach ($foo as &$property) {
+        var_dump($property);
+    }
+} catch (Error $exception) {
+    echo $exception->getMessage() . "\n";
+}
+
+?>
+--EXPECTF--
+Notice: Accessing static property Foo::$property1 as non static in %s:%d
+Cannot acquire reference on final static property Foo::$property1
```

Comment:
I think this test is already working correctly. The error message is not super clear about it, but what happens here is that `$this->property1 = 1` actually creates a dynamic instance property on the object, which is unrelated to the static property declaration and as such does not enforce final.

Reply:
Oh, nice! I didn't know what happens in this case.

Reply:
Nope. I didn't pay attention to `$this->property1 = 1`... And I wanted to use `self::$property1 = 1` Then it's all clear, I just didn't double checked it.